sdk/program/src/message.rs

```diff
@@ -280,6 +280,16 @@ impl Message {
         bincode::serialize(self).unwrap()
     }
 
+    pub fn program_id(&self, instruction_index: usize) -> Option<&Pubkey> {
```

Comment:
These SDK changes could be in a precursor PR too

Reply:
Yup, same as above